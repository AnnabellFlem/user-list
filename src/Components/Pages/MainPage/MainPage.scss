@import 'src/Styles/sizes';
@import 'src/Styles/media';

.list {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0 auto;
}
