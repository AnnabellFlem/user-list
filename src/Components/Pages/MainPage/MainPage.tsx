import React, { useState } from 'react'
import StartPage from '../StartPage'
import GamePage from '../GamePage'
import ResultPage from '../ResultPage'

const MainPage: React.FC = () => {
  const [startGame, setStartGame] = useState(true)
  const [endGame, setEndGame] = useState(false)
  const [win, setWin] = useState(false)
  const [currentNumberQuestion, setCurrentNumberQuestion] = useState(0)

  const handleClick = (
    e: React.MouseEvent<HTMLButtonElement | HTMLElement, MouseEvent>,
  ) => {
    const target = e.currentTarget as HTMLButtonElement
    if (target.value === 'start') {
      setStartGame(false)
    } else if (target.value === 'retry') {
      setCurrentNumberQuestion(0)
      setEndGame(false)
    } else return null
  }

  const handleEndGame = (isEndGame: boolean) => {
    setEndGame(true)
    setWin(isEndGame)
  }

  const handleQuestion = (question: number) => {
    setCurrentNumberQuestion(question)
  }

  return (
    <>
      {startGame && !endGame && <StartPage onButtonClick={handleClick} />}
      {!startGame && !endGame && (
        <GamePage
          handleQuestion={handleQuestion}
          handleEndGame={handleEndGame}
          currentNumberQuestion={currentNumberQuestion}
        />
      )}
      {!startGame && endGame && (
        <ResultPage
          currentNumberQuestion={currentNumberQuestion}
          onButtonClick={handleClick}
          win={win}
        />
      )}
    </>
  )
}

export default MainPage
