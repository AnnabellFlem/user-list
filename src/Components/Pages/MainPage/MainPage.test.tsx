import React from 'react'
import { mount } from 'enzyme'
import MainPage from './MainPage'

describe('<MainPage /> test', () => {
  let wrapper: any

  beforeAll(() => {
    wrapper = mount(<MainPage />)
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Component should render without crashing', () => {
    expect(wrapper.exists()).toBeTruthy()
  })
})
