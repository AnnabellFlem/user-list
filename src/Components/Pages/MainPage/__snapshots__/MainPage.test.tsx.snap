// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> test should to match snapshot 1`] = `ReactWrapper {}`;
